### PR TITLE
chore(deps): bump go-redis v8 to v9

### PR DESCRIPTION
**Pull request by exuan**

### CODE REVIEW

**Comment by yangbodong22011** [comment]
@exuan Thanks for your contribution.
but redis-go v8 and v9 are not compatible https://github.com/redis/go-redis/issues/2085, and a large number of our users are still on Redis6.x, we cannot make this upgrade for now.

**Comment by exuan** [comment]
> @exuan Thanks for your contribution. but redis-go v8 and v9 are not compatible [redis/go-redis#2085](https://github.com/redis/go-redis/issues/2085), and a large number of our users are still on Redis6.x, we cannot make this upgrade for now.
I get it.

I see that the pr https://github.com/redis/go-redis/pull/2110  has been resolved. But a large number of Redis6.x users are still troubled 😢.

**Comment by yangbodong22011** [comment]
@exuan yes, maybe tair-go should also consider releasing two versions based on Redis v8 and v9. This kind of incompatible upgrade is always a headache.

**Comment by exuan** [comment]
> 

This is indeed the current solution, I will continue to pr.

**Comment by yangbodong22011** [comment]
@exuan Before you modify the code, let's discuss it first, such as how to maintain the branch and what package to release.

**Comment by exuan** [comment]
> @exuan Before you modify the code, let's discuss it first, such as how to maintain the branch and what package to release.
It's my pleasure, and I suggest referring to [Conventional Commits](https://www.conventionalcommits.org).

**Comment by ashtonian** [comment]
redis-go v9 is required to connect to a redis v8 cluster, without the change the client won't be able to support redis v8.

File: tair/taircommands.go [hunk 1]
```diff
@@ -121,7 +121,7 @@ type TairCmdable interface {
 	ExHStrLen(ctx context.Context, key, field string) *redis.IntCmd
 	ExHKeys(ctx context.Context, key string) *redis.StringSliceCmd
 	ExHVals(ctx context.Context, key string) *redis.StringSliceCmd
-	ExHGetAll(ctx context.Context, key string) *redis.StringStringMapCmd
```
Comment (yangbodong22011) [inline tair/taircommands.go:128]: breaking change, we don't wanna it.

**Review by yangbodong22011 — COMMENTED** [review]
Everything is OK except the comments. Would you like to rebase develop and continue @exuan

File: tair/taircommands.go (hunk 1 above)
Comment (exuan) [reply tair/taircommands.go:128]: It's my pleasure

**Pull request merged**